modules/investigating-kernel-crashes.adoc

```diff
@@ -0,0 +1,105 @@
++
+[source, terminal]
+----
+sudo rpm-ostree kargs --append='crashkernel=256M'
+----
+
+.  By default, the path in which the vmcore will be saved is `/var/crash`. It is also possible to write the dump over the network or to some other location on the local system by editing `/etc/kdump.conf`. For example, assuming `/var/usrlocal/cores` exists, enter the following command to edit `/etc/kdump.conf` to save the vmcore to `/var/usrlocal/cores`:
++
+[source, terminal]
+----
+sed -i "s/^path.*/path \/var\/usrlocal\/cores/" /etc/kdump.conf
+----
++
+For additional information, see `kdump.conf`, a manual page for the `/etc/kdump.conf` configuration file containing the full documentation of available options, and the comments in `/etc/kdump.conf` and `/etc/sysconfig/kdump`.
+ifdef::openshift-enterprise[]
+Also refer to the link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/system_design_guide/installing-and-configuring-kdump_system-design-guide#configuring-the-kdump-target_configuring-kdump-on-the-command-line[RHEL `kdump` documentation] for further information on configuring the dump target.
+endif::[]
++
+
+ifdef::openshift-enterprise[]
+.  Because `kdump` has trouble finding the correct bootimage location on {op-system}, the `KDUMP_BOOTDIR` variable must be manually set in `/etc/kdump.conf`. You can use `/proc/cmdline` to figure out the `ostree` boot location. For example:
++
+[source, terminal]
+----
+BOOT_LOC=/boot$(cat /proc/cmdline | egrep -o "/ostree/.*/vmlinuz" | sed -e "s|/vmlinuz||g")
+sudo sed -i "s|^#KDUMP_BOOTDIR=\"/boot\"|KDUMP_BOOTDIR=\"${BOOT_LOC}\"|" /etc/sysconfig/kdump
+----
+
+.  Configure `/etc/sysconfig/kdump` to not use the default link:https://man7.org/linux/man-pages/man2/kexec_file_load.2.html[file-based `kexec` syscall] for loading the crash kernel.
++
+[source, terminal]
+----
+sudo sed -i "s|^KEXEC_ARGS=\"-s\"|KEXEC_ARGS=\"\"|" /etc/sysconfig/kdump
+----
+endif::[]
+
+.  Enable the `kdump` systemd service.
++
+[source, terminal]
+----
+sudo systemctl enable kdump.service
+----
+
+.  Reboot your system.
++
+[source, terminal]
+----
+sudo systemctl reboot
+----
+
+.  Ensure that `kdump` has loaded a crash kernel by checking that `kdump.service` has started and exited successfully and that `cat /sys/kernel/kexec_crash_loaded` prints `1`.
+
+== Enabling kdump on day-1
+`kdump` is intended to be enabled per-node to debug kernel problems. It is not recommended to enable `kdump` on all of your nodes in the cluster. Although machine-specific `MachineConfigs` are not yet supported, it is possible to do the above through a systemd unit in a `MachineConfig` object on day-1 and have kdump enabled on all nodes in the cluster. You can create a `MachineConfig` object and inject that object into the set of manifest files used by Ignition during cluster setup. See "Customizing nodes" in the _Installing -> Installation configuration_ section for more information and examples on how to use Ignition configs.
```

Comment:
A few suggestions in this paragraph:
- So that the sentence doesn't start lowercase: **s/`kdump` is intended/The `kdump` service is intended**/
- s/Although machine-specific `MachineConfigs`/Although machine-specific machine configs/ (do not pluralize an object ref, according to docs [guidelines](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#object-references))
- s/it is possible to do the above through a systemd unit/it is possible to do the above through a systemd unit/